pkg/azurefile/azurefile.go

```diff
@@ -371,27 +371,36 @@ func (d *Driver) getFileShareQuota(resourceGroupName, accountName, fileShareName
 }
 
 // get file share info according to volume id, e.g.
-// input: "rg#f5713de20cde511e8ba4900#fileShareName#diskname.vhd#uuid#namespace"
-// output: rg, f5713de20cde511e8ba4900, fileShareName, diskname.vhd, namespace
-func GetFileShareInfo(id string) (string, string, string, string, string, error) {
+// input: "rg#f5713de20cde511e8ba4900#fileShareName#diskname.vhd#uuid#namespace#subsID"
+// output: rg, f5713de20cde511e8ba4900, fileShareName, diskname.vhd, namespace, subsID
+func GetFileShareInfo(id string) (string, string, string, string, string, string, error) {
 	segments := strings.Split(id, separator)
 	if len(segments) < 3 {
-		return "", "", "", "", "", fmt.Errorf("error parsing volume id: %q, should at least contain two #", id)
+		return "", "", "", "", "", "", fmt.Errorf("error parsing volume id: %q, should at least contain two #", id)
 	}
 	rg := segments[0]
-	var diskName, namespace string
+	var diskName, namespace, subsID string
 	if len(segments) > 3 {
 		diskName = segments[3]
 	}
-	if len(segments) > 4 && rg == "" {
+	if rg == "" {
 		// in csi migration, rg could be empty, then the 5th element is namespace
 		// https://github.com/kubernetes/kubernetes/blob/v1.23.5/staging/src/k8s.io/csi-translation-lib/plugins/azure_file.go#L137
-		namespace = segments[4]
-	}
-	if len(segments) > 5 {
-		namespace = segments[5]
+		if len(segments) > 4 {
+			namespace = segments[4]
+		}
+		if len(segments) > 5 {
```

Comment:
should be like this:
```
	if rg == "" {
		// in csi migration, rg could be empty, then the 5th element is namespace
		// https://github.com/kubernetes/kubernetes/blob/v1.23.5/staging/src/k8s.io/csi-translation-lib/plugins/azure_file.go#L137
		if len(segments) > 4 {
			namespace = segments[4]
		}
	} else {
		if len(segments) > 5 {
			namespace = segments[5]
		}
	}
	if len(segments) > 6 {
		subsID = segments[6]
	}
```

Reply:
also write a ut for `rg == ""` condition

Reply:
csi migration does not support cross subs, we don't need to consider such scenario

Reply:
fixed it, and there is already an ut covered `rg == ""` condition